gsadjust/pyqt_models.py

```diff
@@ -1260,115 +1316,79 @@ def data(self, index, role=QtCore.Qt.DisplayRole):
-                                item = loop.child(i, 0)
-                                if item.station_name == old_name:
-                                    item.station_name = new_name
-                                    for iiii in range(len(item.station)):
-                                        item.station[iiii] = new_name
-                        if rename_type == 'Survey':
-                            loop = m.parent()
-                            survey = loop.parent()
-                            for i in range(survey.rowCount()):
-                                loop = survey.child(i, 0)
-                                for ii in range(loop.rowCount()):
-                                    item = loop.child(ii, 0)
-                                    if item.station_name == old_name:
-                                        item.station_name = new_name
-                                        for iiii in range(len(item.station)):
-                                            item.station[iiii] = new_name
-                        if rename_type == 'Campaign':
-                            campaign = index.model().invisibleRootItem()
-                            for i in range(campaign.rowCount()):
-                                survey = campaign.child(i, 0)
-                                for ii in range(survey.rowCount()):
-                                    loop = survey.child(ii, 0)
-                                    for iii in range(loop.rowCount()):
-                                        item = loop.child(iii, 0)
-                                        if item.station_name == old_name:
-                                            item.station_name = new_name
-                                            for iiii in range(len(item.station)):
-                                                item.station[iiii] = new_name
-                        if rename_type == 'Station':
-                            m.station_name = new_name
-                            for iiii in range(len(m.station)):
-                                m.station[iiii] = new_name
-                        logging.info('Stations renamed from {} to {} in {}'.format(old_name, new_name,
-                                                                                    rename_type))
-                        self.signal_name_changed.emit()
-
-                elif type(m) is ObsTreeLoop:
-                    new_name = str(value)
-                    old_name = m.name
-                    logging.info('Loop renamed from {} to {}'.format(old_name, new_name))
-                    m.name = new_name
-            return True
+            if index.isValid() and index.column() == 0:
+                item = index.model().itemFromIndex(index)
+                return {
+                    ObsTreeStation: self._handler_edit_ObsTreeStation,
+                    ObsTreeLoop: self._handler_edit_ObsTreeLoop
+                }[type(item)](item, value)
```

Comment:
This takes the map-routing approach a step further, by defining specific `_handler` methods and routing to them based on the type of the item. We pass in the item and value and then these functions do the rest.

For readability, it's generally a good idea to try and isolate blocks like the handling code in their own methods rather than under `if` branches.

You can take this a step simpler, retaining the if and then branching to the relevant handler instead.

Reply:
very nice